test/flatten/flatten.c

```diff
@@ -161,6 +129,63 @@ int main(int argc, char **argv)
 
     DTP_pool_free(dtp);
 
+    return NULL;
+}
+
+int main(int argc, char **argv)
+{
+    int num_threads = 1;
+
+    while (--argc && ++argv) {
+        if (!strcmp(*argv, "-datatype")) {
+            --argc;
+            ++argv;
+            strncpy(typestr, *argv, MAX_DTP_BASESTRLEN);
+        } else if (!strcmp(*argv, "-count")) {
+            --argc;
+            ++argv;
+            basecount = atoi(*argv);
+        } else if (!strcmp(*argv, "-seed")) {
+            --argc;
+            ++argv;
+            seed = atoi(*argv);
+        } else if (!strcmp(*argv, "-iters")) {
+            --argc;
+            ++argv;
+            iters = atoi(*argv);
+        } else if (!strcmp(*argv, "-verbose")) {
+            verbose = 1;
+        } else if (!strcmp(*argv, "-num-threads")) {
+            --argc;
+            ++argv;
+            num_threads = atoi(*argv);
+        } else {
+            fprintf(stderr, "unknown argument %s\n", *argv);
+            exit(1);
+        }
+    }
+    if (typestr == NULL || basecount <= 0 || seed < 0 || iters < 0 || num_threads < 0) {
```

Comment:
Should this check be for `num_threads <= 0`?
Same comment for `iters`

Reply:
Fixed.